### ISSUE TITLE
"use strict" and `this` use inside constructor of a child class

### ISSUE THREAD

**Issue by jhorwit2** [babel/babel#1155]
So, I was upgrading babel and noticed a weird use case that was yelling at me when it seems like it shouldn't. I have global 'use strict' in the file. 

here is a quick blurb that should show the issue:

```javascript
constructor (options) {
   let parentOptions = {};
    // do some stuff to prepare options for parent super
    parentOptions.init = function () {
        options.init.call(this);
        // do other stuff
    };
    // do some more stuff perhaps
   super(parentOptions);
}
```

The above example yells at me because I'm using `this` prior to `super` being called; however, with global 'use strict' the function where I call `this` doesn't actually apply to the `this` context of the class itself. Instead, it uses the context of the method invocation using call, apply, or bind. 

The exact file can be found here, if curious. https://github.com/addthis/fluxthis/blob/master/src/ImmutableStore.es6.js

**Comment by jhorwit2** [comment]
And this is why we <3 this project.

**Comment by sebmck** [comment]
Sorry for the delay, fixed as of 5.0.8. Thanks!
